lambdas/postcode-lookup/src/test/java/uk/gov/di/ipv/cri/address/api/handler/PostcodeLookupHanderTest.java

```diff
@@ -128,6 +129,38 @@ void AnyOtherExceptionThrows500()
         verifyNoMoreInteractions(eventProbe);
     }
 
+    @Test
+    void PostcodeLookupTimeoutExceptionThrows504()
+            throws PostcodeLookupValidationException, PostcodeLookupProcessingException,
+                    SessionExpiredException, SessionValidationException, SessionNotFoundException,
+                    JsonProcessingException {
+        String testPostcode = "LS1 1BA";
+        String sessionId = String.valueOf(UUID.randomUUID());
+        Map<String, String> requestHeaders = Map.of("session_id", sessionId);
+        SessionItem sessionItem = mock(SessionItem.class);
+        AuditEventContext testAuditEventContext = mock(AuditEventContext.class);
+
+        PostcodeLookupTimeoutException exception =
+                new PostcodeLookupTimeoutException("Error Connection Timeout");
+
+        when(eventProbe.log(INFO, "found session")).thenReturn(eventProbe);
+        setupEventProbeExpectedErrorBehaviour();
+
+        when(apiGatewayProxyRequestEvent.getHeaders()).thenReturn(requestHeaders);
+        when(apiGatewayProxyRequestEvent.getPathParameters())
+                .thenReturn(Map.of("postcode", testPostcode));
+        when(sessionService.validateSessionId(sessionId)).thenReturn(sessionItem);
+        when(postcodeLookupService.getAuditEventContext(testPostcode, requestHeaders, sessionItem))
+                .thenReturn(testAuditEventContext);
+        willThrow(exception).given(postcodeLookupService).lookupPostcode(anyString());
```

Comment:
Please could you change to from BDD to TDD style `doThrow(exception).when(postcodeLookupService).lookupPostcode(anyString());
`
just for consistency, this isn't you I could see some it else where